File: stlm-java/src/smoothing/AdditiveSmoothing.py
#!/usr/bin/env python
""" generated source for module AdditiveSmoothing """
# package: smoothing
import segment.Segment

import suffixtree.SuffixTree

class AdditiveSmoothing(Smoothing, T):
    """ generated source for class AdditiveSmoothing """
    disc = 1.0

    def __init__(self, t):
        """ generated source for method __init__ """
        super(AdditiveSmoothing, self).__init__(t)

    def wordHereMultChild(self, parent, child):
        """ generated source for method wordHereMultChild """
        prob = (self.disc + child.getCount()) / (parent.numChildren() * self.disc + parent.getCount())
        return prob

    def updateUnk(self):
        """ generated source for method updateUnk """
        unk = self.disc / (current.numChildren() * self.disc + current.getCount())

File: stlm-java/src/smoothing/AdditiveSmoothing.java
package smoothing;

import segment.Segment;
import suffixtree.SuffixTree;

public class AdditiveSmoothing<T> extends Smoothing<T> {
	
	double disc = 1.0;

	public AdditiveSmoothing(SuffixTree<T> t) {
		super(t);
	}
	
    public double wordHereMultChild(Segment parent, Segment child) {
    	double prob = (disc + child.getCount()) / (parent.numChildren() * disc + parent.getCount()); 
    	
		return prob;
	}
    
    public void updateUnk() {
    	unk = disc / (current.numChildren() * disc + current.getCount()); 
    }
	
}
